### DISCUSSION TITLE
Read only mode?

### DISCUSSION THREAD

**Discussion by Toremok** [jgraph/drawio-desktop#1380] [Q&A]
Hi all,

I'm using the desktop version of draw.io / diagrams.net and I'm not sure how to enter a read-only view for the files.
I have many diagrams with hyperlinks etc., and I want to be able to put files in a "preview" state where it can be viewed, open links, allow text copying, etc. but without the ability to make any changes. I've seen recommendation to lock the diagrams, but then if I use hyperlinks it shows me a "save changes" message at the top. My goal is to interact with the diagram without the potential to make changes that modify or delete items.

I'd really appreciate any help.

**Comment by davidjgraph** [comment]
That doesn't really make sense for the desktop version. You'd use the online version for sharing.

**Reply by Toremok** [reply]
Hey David, thanks for responding. Can you clarify what about it doesn't make sense? I must be conceptualizing viewing the files much differently than you. What is the advantage of using the online version for a local file?
For my needs, I don't have any intention of sharing the diagram, just viewing it.

**Reply by Toremok** [reply]
I saw your Expectations post from last week so I don't want to overstep the boundary and if the final answer is "this type of feature is not a concern" I respect the decision. That being said, I simply don't understand how it "doesn't make sense." To me it seems very logical to have an option of _viewing_ a flowchart as opposed to _editing_ one. I'm writing once again as this has been a major gripe for a while now, to the point where i'm considering finding a different method of organizing many files of information. The canvas/graph style of expansive diagrams makes a "viewing" mode much more relevant than for text layouts. I see the desktop application as a tool that gives me a happy medium between a graph database and a word document, all on a local machine. When files have hyperlinks, different shapes and sizes, and text all over the screen, being able to lock down the information presented seems like a necessary feature.

I hope this post isn't rude, I'm just so surprised this isn't a standard option that i'm second-guessing how i've been using the software. If anyone wants to explain how they use their diagrams, with any potential advice for this kind of issue, i'd really appreciate it. I'm an edit-and-a-half away from giving up and laying out my files in a different way entirely. Thanks!

**Reply by davidjgraph** [reply]
I guess it depends if you grew up in a SaaS world or not. If you look at many open source apps, Inkscape, libreoffice, gimp, they follow this old school model. Modern apps like MS office will connect to cloud storage, it's true.

We're targeting the privacy centric users, the aim is a completely locked down app with zero external connections.

We also don't run any backend storage, that would be required for the model you describe.

What you can do is save with a .png suffix in the app (save, not export). Then you have a ready-made viewable file that also contains the saved diagram that you can re-load.

**Reply by Toremok** [reply]
I'm also a privacy centric user, which is why using an online version is a no-go for me. 
Unfortunately the .png solution makes the diagram an image, rendering hyperlinks unusable. Regardless, I appreciate the informative reply and the idea for a potential workout. 

I'll see if I can come to terms with using it in this way. If not, it may make more sense to reorganize the files to a more text-like format.
Kudos on all the work you put in!

**Reply by davidjgraph** [reply]
I'm afraid I just don't understand the paradigm then. Do you have an example with a similar productivity app?

**Reply by Toremok** [reply]
I don't and maybe that's actually the problem, but then again I don't really know many productivity apps.

I mainly use orgmode and org-roam for plain-text and general (though simple) organization. Openoffice for more specific and one-off documents. 

For a while now i've been moving certain plain-text "pages" into the draw.io program to have a more holistic view of related information. Instead of using org-roam for a large graph, I have pieces of information that are more localized in draw.io on one "sheet", with hyperlinks as quick buttons to "open" and "close" relevant entries. For example: if I have a sheet called "Animal Kingdom" I may create different bubbles of information based on taxonomy, with hyperlinks to smaller shapes that contain text about specific animals within each genus. 

The main idea is to present text in a way that follows a process (the flowchart aspect) while still allowing for information to be seen/hidden (hyperlinks) all while maintaining a relatively static presentation (similar to plain-text). For this reason, being able to save and "lock" the information is very important to me. There's a lot of info and since I click around the sheet I want to be sure i'm not making accidental modifications or adding/removing anything. A preview page would allow me to move around the sheet freely, allow me to click hyperlinks, and yet not change the actually content of the sheet itself. 

I hope that made sense. If you have any suggestions for such a software, or even just software in general you recommend, I'll look into them. Thanks!

**Reply by tmal-1** [reply]
I agree with the original requestor and can provide an example.

The eEd graph editor is available both in desktop and online versions. It supports two different modes, Edit Mode and Navigation Mode. In Edit Mode the full functionality of the editor is available; in Navigation Mode you can explore a graph but not make changes.

One use for this is to create a diagram with several pages and hyperlinks between them. For example, it is convenient in a flowchart to create a link from a subprocess symbol to the page that describes the subprocess. In Navigation Mode you can explore the flowchart and follow links without needing to worry that you make accidental changes.

Consequently, I would like to second this request. Please can draw.io support a mode similar to yEd's Navigation Mode.

**Reply by rossnet** [reply]
There are other tools as well that support read-only / view modes. 

I actually think most browser based editing tools support view modes to allow sharing their files with users that do not get/have write permissions.

E.g. when you use Collabora Office (just to name one) integrated with Nextcloud: If you share a file, you can set the permission to "read only" causing the user the file is shared with to get the file rendered in the view mode of Collabora.

I don't think this is about desktop vs. online version of draw.io as to my understanding both of them do not support a "view mode"? Or is there a view mode available for the online version and if so, how do I activate it?

**Comment by mattiamarchesini** [comment]
"File/project sharing" and a local "read only mode" are and should be two different use cases, I'm pretty sure we can all agree with that.

While I can see why this feature wouldn't be an immediate priority right now, the "use the web version" **workaround** you are suggesting is just that: a workaround. Not to mention I see 5 main flaws on why this is not ideal:

- The changes made in the diagrams always take some amount of delay before also being present in the sharing version, but a local view mode would be immediate without having to go to a different browser tab every time
- Actually a lot of apps that deal with diagrams and similar seem to see the value of a "read only mode". Obsidian canvas just to name one.
- I'm tired of looking at my diagrams to see if everything is ok and then move something by accident. Happens every time plz help 😢 
- Especially when you use a touch screen to move around. At that point a read only mode becomes pretty much indispensable.
- I personally think the desktop and web version should be as compatible in terms of features as they can be, and this is a step in the wrong direction for a feature that wouldn't be that big of a deal to implement

**Comment by davidjgraph** [comment]
Try 12.4.0

**Reply by Toremok** [reply]
Excuse the ignorance - can you explain what 12.4.0 means?
i've downloaded the latest draw.io desktop version. Does it include the option for a read only mode?

Thanks for remaining active in this discussion.

**Reply by mattiamarchesini** [reply]
To this day, locking down all the layers is the only way to actually do it, even if the "unsaved changes" banner will still appear each time you lock/unlock (not like it will disturb you it's very small).

It works like the protected view in Microsoft PowerPoint, in the sense that the "locking" is an integral part of the file metadata and a locked diagram that you share with someone will appear locked to him/her as well.

Because of this, I don't think we will see any changes in this regard any time soon, because the feature is actually there, but not exactly the way we wanted.

**Reply by alderg** [reply]
The feature is in v24.5.1 online via File, Properties and will be in the next desktop release. Meanwhile you can also make the file read-only on the filesystem.

**Reply by Toremok** [reply]
Just to be sure, i've downloaded the newest desktop release from a few days ago. I see that under 
`File --> Properties`
the first option is "Locked" which you can then put a checkmark on and click apply. 

As far as I can tell, this locks the sheet. It still allows for hyperlinks to be toggled on/off (and interestingly, still changes the status to "saving..." and then "saved", but there is now a visual cue of a closed lock to make clear the sheet cannot be edited). I also noticed ticking the checkmark for the "Locked" option applies the lock to the entire sheet, with no option to do it on a per page basis.

I cannot speak for anyone else, but this provides a very good solution to what I was after. Kudos to the team for being receptive, engaging, and considerate of user requests!
If there's a place to read more in depth about this feature and it's properties (or if it can be written out more explicitly here) I'd love to read up on it.

Again, thank you!
